pkg/publicapi/middleware/version.go

```diff
@@ -70,10 +79,18 @@ func VersionNotifyLogger(logger *zerolog.Logger, serverVersion semver.Version) e
 				notif.Message = fmt.Sprintf("received request with invalid client version: %s", cVersion[0])
 				return nil
 			}
+
+			// Strip the pre-release tag
+			strippedClientVersion, err := stripPreRelease(clientVersion)
+			if err != nil {
+				notif.Message = fmt.Sprintf("error stripping pre-release tag from client version: %s", err)
+				return nil
+			}
+
 			// extract parsed client version for comparison
-			notif.ClientVersion = clientVersion.String()
+			notif.ClientVersion = strippedClientVersion.String()
 
-			diff := serverVersion.Compare(clientVersion)
+			diff := serverVersion.Compare(strippedClientVersion)
```

Comment:
we don't want to strip the client version here, this method is for logging the version of the request received from the client. Please revert this change.